addons/website_sale/controllers/main.py

```diff
@@ -1455,6 +1455,10 @@ def process_express_checkout(
             if shipping_option:
                 order_sudo._check_carrier_quotation(force_carrier_id=int(shipping_option['id']))
 
+            # partner for shipping may have missing phone number
+            if not order_sudo.partner_shipping_id.phone and order_sudo.partner_invoice_id.phone:
+                order_sudo.partner_shipping_id.phone = order_sudo.partner_invoice_id.phone
+
```

Comment:
This won't update the phone number of the shipping address if there is already one.